backend/infrahub/cli/dev.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+import importlib
+import logging
+from pathlib import Path  # noqa: TC003
+from typing import TYPE_CHECKING
+
+import typer
+from graphql import parse, print_ast, print_schema
+from infrahub_sdk.async_typer import AsyncTyper
+from rich.logging import RichHandler
+
+from infrahub import config
+from infrahub.core.initialization import (
+    first_time_initialization,
+    initialization,
+)
+from infrahub.core.schema import SchemaRoot, core_models, internal_schema
+from infrahub.core.schema.schema_branch import SchemaBranch
+from infrahub.core.utils import delete_all_nodes
+from infrahub.graphql.manager import GraphQLSchemaManager
+from infrahub.graphql.schema_sort import sort_schema_ast
+from infrahub.log import get_logger
+
+if TYPE_CHECKING:
+    from infrahub.cli.context import CliContext
+
+app = AsyncTyper()
+
+
+@app.command(name="export-graphql-schema")
+async def export_graphql_schema(
+    ctx: typer.Context,  # noqa: ARG001
+    config_file: str = typer.Option("infrahub.toml", envvar="INFRAHUB_CONFIG"),
+    out: Path = typer.Option("schema.graphql"),  # noqa: B008
+) -> None:
+    """Export the GraphQL schema to a file."""
```

Comment:
For clarity I think we should still say "Export the core GraphQL schema to a file."